src/target/riscv/riscv.h

```diff
@@ -193,7 +193,18 @@ struct riscv_info {
 	 * was resumed. */
 	int (*resume_go)(struct target *target);
 	int (*step_current_hart)(struct target *target);
-	int (*on_halt)(struct target *target);
+
+	/* These get called from riscv_poll_hart(), which is a house of cards
+	 * together with openocd_poll(), so be careful not to upset things too
+	 * much. */
+	int (*became_halted)(struct target *target);
+	int (*became_running)(struct target *target);
+	int (*became_unavailable)(struct target *target);
```

Comment:
Could we call these `handle_became_*`? To make it very clear from the name that these are handlers/callback, and not for instance just bool flags.

```suggestion
	int (*handle_became_halted)(struct target *target);
	int (*handle_became_running)(struct target *target);
	int (*handle_became_unavailable)(struct target *target);
```